### PR TITLE
feat: extract shell scripts (IN-304)

### PR DESCRIPTION
**Pull request by Xavientois**
**Fixes or implements IN-304**

### Brief description. What is this change?

Extracts longer shell scripts into external files to allow for reuse of scripts (like `scripts/e2e/setup-database.sh`).

This will also allow for linting of shell scripts.

### CODE REVIEW

**Review by xavidop — CHANGES_REQUESTED** [review]
is this finished yet? I am missing a bunch of shell scripts. I don't like too much the current structure. I would like to have the scripts in the same directory as the command with the same name. Example, down src/jobs/cdk:
1. cdk-deploy.yml
2. cdk-deploy.sh

File: src/scripts/utils/exec-with-retry.sh [hunk 1]
```diff
@@ -5,9 +5,7 @@ retry() {
     n=0
     until [ $n -ge $MAX_RETRY ]; do
     set +e
-    set -x
```
Comment (xavidop) [inline src/scripts/utils/exec-with-retry.sh]: why are you enabling debugging here?

**Review by xavidop — CHANGES_REQUESTED** [review]

File: src/scripts/utils/exec-with-retry.sh (hunk 1 above)
Comment (Xavientois) [reply src/scripts/utils/exec-with-retry.sh]: Oops, I thought I removed that. I put it there when I was debugging it yesterday

File: src/commands/docker/build_push_image.yml [hunk 2]
```diff
@@ -69,71 +69,18 @@ steps:
...
-            --build-arg NPM_TOKEN=//registry.npmjs.org/:_authToken=${NPM_TOKEN} \
-          <</ parameters.monorepo_directory >> \
-          --build-arg build_BUILD_NUM=${CIRCLE_BUILD_NUM} \
-          --build-arg build_BUILD_URL=${CIRCLE_BUILD_URL}	\
-          --build-arg build_GITHUB_TOKEN=${GITHUB_TOKEN} \
-          --build-arg build_GIT_SHA=${CIRCLE_SHA1} \
-          --build-arg build_SEM_VER=${SEM_VER} \
-          -f << parameters.build_context >>/<< parameters.dockerfile >> \
-          -t $IMAGE_NAME << parameters.build_context >>
+      environment:
```
Comment (Xavientois) [inline src/commands/docker/build_push_image.yml:72]: here

**Comment by Xavientois** [comment]
Closing this in favour of making these changes incrementally

**Pull request closed without merging**